### ISSUE TITLE
Test: Webview persistence APIs

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/vscode#50696]
Test for #49022 #48542

Complexity: 3

- [x] Any OS @octref
- [ ] Any OS @rebornix 
 
**Summary**
A new api for webview serialization has been added. This allows webviews to both save their state between becoming hidden and visible, as well as saving their state when VS Code restarts

**Test**
Try adding the two types of persistence to the new webview (you can use the [cat coding example](https://github.com/Microsoft/vscode-extension-samples/blob/master/webview-sample/) for this)

Visibility persistence is implemented using a `getState` and `setState` method inside the webview itself:

```ts
const vscode = acquireVsCodeApi();
const state = vscode.getState();
vscode.setState({ abc: 123 })
```

Serialization uses [`WebviewPanelSerializer`](https://github.com/Microsoft/vscode/blob/f5300ced7fbd0193babf05ee37860bb2d160beb7/src/vs/vscode.d.ts#L5423) and [`registerWebviewPanelSerializer`](https://github.com/Microsoft/vscode/blob/f5300ced7fbd0193babf05ee37860bb2d160beb7/src/vs/vscode.d.ts#L5942) to register a handler in the extension that can revive a webview from its state.

**Things to test:**

* Can you implement the two types of persistence?
* Does the API documentation make sense? (I'll be writing docs for it too but have not finished these yet)
* Are there cases where the new APIs may not work?

**Comment by rebornix** [comment]
I have a react app (which is basically a tree view) running in the webview, tried to save/restore the state of tree expand/collapse state when the webview become visible/invisbile. Below are my attempts and findings

* It took me a while to figure out how to use `setState`, I was expecting things @dbaeumer mentioned in #48542, get an event for webview visibility change and post a message back to the webview and finally `setState` in the webview message handler. Turns out once the webview is invisible there is no way to post message to it. So I need to think of a way to store the view state when a bit more eagerly.
* Then I tried to save React component's state when it's being changed. As I want to store states of all components, I tried to read the state for vscode storage first, apply changes to it and then save. 

    ```
    const state = vscode.getState()
    vscode.setState({}, state, {
        componentId: expand
    })
    ```
  unluckily, the state I get is always `undefined`. It seems the state is only saved when the webview becomes invisible.
* I didn't get `WebviewPanelSerializer` working yet. I'm not sure when should I register the seriablizer and when the callbacks will be invoked, especially when my extension is only activated when some commands are triggerred.
